lib/plausible_web/controllers/auth_controller.ex

```diff
@@ -24,52 +24,61 @@ defmodule PlausibleWeb.AuthController do
               :activate_form
             ]
 
-  def register_form(conn, _params) do
-    if Keyword.fetch!(Application.get_env(:plausible, :selfhost), :disable_registration) != false do
-      redirect(conn, to: Routes.auth_path(conn, :login_form))
-    else
-      changeset = Plausible.Auth.User.changeset(%Plausible.Auth.User{})
+  plug :maybe_disable_registration when action in [:register_form, :register]
```

Comment:
I decided not to change any other action, but more actions that check for `configplausible].selfhost.disable_registration` could be added here later